compose/compose-file/index.md

```diff
@@ -1431,6 +1431,18 @@ refer to it within the Compose file:
         external:
           name: actual-name-of-volume
 
+> External volumes are always created with docker stack deploy
+>
+External volumes that do not exist _will be created_ if you use [docker stack
```

Comment:
Sorry for my late reply; I wonder if we should call out that this is not really "by design", but a current "limitation". Basically, it's a bug (or, indeed a current limitation); I think it's important to add "some" note about that. Users should not rely on this auto-creation, because (hopefully) we'll be able to fix it, and the behaviour will change in future.

TL;DR if you define a volume as _external_, you should create it up front.

but ping @cpuguy83 for input as well